sql/core/src/main/java/org/apache/spark/sql/sources/v2/reader/streaming/StreamingReadSupport.java

```diff
@@ -20,11 +20,10 @@
 import org.apache.spark.sql.sources.v2.reader.ReadSupport;
 
 /**
- * A base interface for streaming read support. This is package private and is invisible to data
- * sources. Data sources should implement concrete streaming read support interfaces:
- * {@link MicroBatchReadSupport} or {@link ContinuousReadSupport}.
+ * A base interface for streaming read support. Data sources should implement concrete streaming
+ * read support interfaces: {@link MicroBatchReadSupport} or {@link ContinuousReadSupport}.
  */
-interface StreamingReadSupport extends ReadSupport {
+public interface StreamingReadSupport extends ReadSupport {
```

Comment:
Hi, @cloud-fan .
I know the previous design choice here, but I'm wondering if we can change like this.

Reply:
That's weird; the access is coming from the same package. I wonder why it generates IllegalAccessError? Is there any other way to work around it, like placing some public subinterface in the test source tree that mockito can access?
We have used `@VisibleForTesting` elsewhere to annotate things that are only public for test purposes, FWIW.

Reply:
`VisibleForTesting` is added. And, yes. I agree that this is a weird and totally unexpected situation. I've investigated why this happens in Maven testing only and tried to find some Maven-dependent solutions. Until now, there was no luck.